pose_estimation_ros/src/pose_estimation_ros.cpp

```diff
@@ -0,0 +1,86 @@
+#include <pose_estimation_ros/pose_estimation_ros.hpp>
+
+namespace ariitk::pose_estimation_ros {
+
+void PoseEstimationROS::init(ros::NodeHandle& nh, ros::NodeHandle& nh_private) {
+	detected_platform_odom_pub_ = nh.advertise<nav_msgs::Odometry>("detected_pose", 1);
+	quad_pose_sub_ = nh.subscribe("quad_odometry", 1, &PoseEstimationROS::quadPoseCallBack, this);
+	quad_pixel_coordinates_sub_ = nh.subscribe("platform_centre", 1, &PoseEstimationROS::pixelCoordinatesCallBack, this);
+	is_platform_detected_sub_ = nh.subscribe("platform_status", 1, &PoseEstimationROS::platformStatusCallback, this);
+
+	nh_private.getParam("camera_to_quad_matrix", camera_to_quad_matrix_);
+	nh_private.getParam("camera_matrix/data", camera_matrix_);
+	nh_private.getParam("distortion_coefficients/data", distortion_matrix_);
+	nh_private.getParam("camera_translation", camera_translation_);
+	nh_private.getParam("loop_rate", loop_rate);
+	nh_private.getParam("platform_height", platform_height_);
+
+	scaleUpMatrix = Eigen::Matrix3d::Zero();
+
+	platform_odom_[0].pose.pose.position.x = 0;
+	platform_odom_[0].pose.pose.position.y = 0;
```

Comment:
better impl: `platform_odom_[0].pose.pose.position = {0.0, 0.0, 0.0}`

Reply:
This gives build errors.

Reply:
yeah, i wasn't sure if it would work. no prob